app/tests/unit/components/acronyms.spec.js

```diff
@@ -55,6 +56,110 @@ describe('getAcronym', () => {
   });
 });
 
+describe('getUsers', () => {
+  const acronymUsersFixture = require('./fixtures/acronymUsers.json');
+  const keycloakUsersFixture = require('./fixtures/keycloakUsers.json');
+
+  // Spy/mock the DB access 'find' method on the acronym table
+  const spy = jest.spyOn(acronymService, 'acronymUserList');
+  const usersSpy = jest.spyOn(usersComponent, 'getAllGetokUsers');
+
+  beforeEach(() => {
+    spy.mockClear();
+    usersSpy.mockClear();
+  });
+
+  it('should throw an error if no acronym supplied', async () => {
+    await expect(acronyms.getUsers(undefined)).rejects.toThrow('No app acronym supplied to getUsers');
+  });
+
+
+  it('should return an empty array if the service returns undefined', async () => {
+    acronymService.acronymUserList.mockResolvedValue(undefined);
+    const acr = 'XXX';
+
+    const result = await acronyms.getUsers(acr);
+
+    expect(result).toEqual([]);
+    expect(spy).toHaveBeenCalledTimes(1);
+    expect(spy).toHaveBeenCalledWith(acr);
+    expect(usersSpy).toHaveBeenCalledTimes(0);
+  });
+
+  it('should return an empty array if no userAcronym mappings exist', async () => {
+    acronymService.acronymUserList.mockResolvedValue([]);
+    const acr = 'XXX';
+
+    const result = await acronyms.getUsers(acr);
+
+    expect(result).toEqual([]);
+    expect(spy).toHaveBeenCalledTimes(1);
+    expect(spy).toHaveBeenCalledWith(acr);
+    expect(usersSpy).toHaveBeenCalledTimes(0);
+  });
+
+  it('should return an empty array if no users are found from KC', async () => {
+    acronymService.acronymUserList.mockResolvedValue(acronymUsersFixture);
+    usersComponent.getAllGetokUsers.mockResolvedValue([]);
+    const acr = 'XXX';
+
+    const result = acronyms.getUsers(acr);
+    await expect(result).rejects.toThrow(`An error occured fetching users for acronym ${acr}`);
+
+    expect(spy).toHaveBeenCalledTimes(1);
+    expect(spy).toHaveBeenCalledWith(acr);
+    expect(usersSpy).toHaveBeenCalledTimes(1);
+  });
+
+  it('should return an appropriate combination of AcronymUser and Keycloak user records', async () => {
+    acronymService.acronymUserList.mockResolvedValue(acronymUsersFixture);
+    usersComponent.getAllGetokUsers.mockResolvedValue(keycloakUsersFixture);
+    const acr = 'XXX';
+
+    const result = await acronyms.getUsers(acr);
+
+    expect(spy).toHaveBeenCalledTimes(1);
+    expect(spy).toHaveBeenCalledWith(acr);
+    expect(usersSpy).toHaveBeenCalledTimes(1);
+
+    expect(result).toBeTruthy();
+    expect(result.length).toEqual(2);
```

Comment:
Jest has an equivalent helper function for length evaluation: `expect(result).toHaveLength(2);`

Reply:
Cool